compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InlineCallableReferenceToLambda.kt

```diff
@@ -49,191 +45,213 @@ internal val inlineCallableReferenceToLambdaPhase = makeIrFilePhase(
+                referencedFunction.returnType,
+                function.symbol,
+                function.symbol.descriptor,
+                referencedFunction.typeParameters.size,
+                IrStatementOrigin.LAMBDA
+            ).apply {
+                copyAttributes(expression)
             }
-        })
-    }
-
-    private fun functionReferenceToLambda(
-        scope: ScopeWithIr,
-        expression: IrCallableReference,
-        referencedFunction: IrFunction
-    ): IrExpression {
-        val irBuilder =
-            context.createIrBuilder(scope.scope.scopeOwnerSymbol, expression.startOffset, expression.endOffset)
-
-
-        val boundReceiver = expression.dispatchReceiver ?: expression.extensionReceiver
-
-        val expectedType = expression.type
-        var expectedNumValueParameters: Int
-        if (expectedType.isFunctionOrKFunction() || expectedType.isSuspendFunction() || expectedType.isKSuspendFunction()) { // TODO: handle subtypes
-            expectedNumValueParameters = (expectedType as IrSimpleType).arguments.size - 1  // In ...Function classes, the last argument is return type
-            referencedFunction.dispatchReceiverParameter?.let { expectedNumValueParameters-- }
-            referencedFunction.extensionReceiverParameter?.let { expectedNumValueParameters-- }
-            boundReceiver?.let { expectedNumValueParameters++ }
-            assert(referencedFunction.valueParameters.subList(expectedNumValueParameters, referencedFunction.valueParameters.size).all {
-                it.defaultValue != null
-            })
-        } else {
-            assert(expectedType.classOrNull?.isSubtypeOfClass(context.irBuiltIns.kPropertyClass) == true)
-            expectedNumValueParameters = 0
         }
+    }
 
+    private fun expandInlineGetterReferenceToLambda(expression: IrPropertyReference, getter: IrFunction): IrExpression {
+        val irBuilder = context.createJvmIrBuilder(currentScope!!.scope.scopeOwnerSymbol, expression.startOffset, expression.endOffset)
         return irBuilder.irBlock(expression, IrStatementOrigin.LAMBDA) {
-            lateinit var variableForBoundReceiver: IrVariable
-            if (boundReceiver != null) {
-                variableForBoundReceiver = createTmpVariable(boundReceiver, BOUND_REFERENCE_RECEIVER)
-            }
 
-            val newLambda = buildFun {
+            val function = buildFun {
                 setSourceRange(expression)
                 origin = JvmLoweredDeclarationOrigin.FUNCTION_REFERENCE_IMPL
                 name = Name.identifier("stub_for_inline")
                 visibility = Visibilities.LOCAL
-                returnType = referencedFunction.returnType
+                returnType = getter.returnType
                 isSuspend = false
             }.apply {
-                if (referencedFunction is IrConstructor) {
-                    copyTypeParametersFrom(referencedFunction.parentAsClass)
+
+                if (getter is IrConstructor) {
```

Comment:
Look like this piece of code should be in 'expandInlineFunctionReferenceToLambda' (and test is required for missed case)
Why did you make separation between `expandInlineGetterReferenceToLambda` and 'expandInlineFunctionReferenceToLambda'?

Reply:
Thank you, will add a test.

Their bodies seemed different enough (varargs, defaults etc) that it seemed clearer to have two separate code paths for the two cases. I am open to suggestions.

Reply:
@max-kammerer Wait, got it. Admittedly it _was_ a long day yesterday :)

Hold for a revised PR.